// vuestic-ui/dist/web-components/src/components/va-file-upload/VaFileUploadUndo/VaFileUploadUndo.js
import { _ as __default__ } from "./VaFileUploadUndo.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, Fragment, createVNode, createElementVNode, normalizeClass, toDisplayString, withCtx, createTextVNode } from "vue";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-file-upload-undo-button-line-height: 1.1428;\n  --va-file-upload-undo-button-margin-top: 0;\n  --va-file-upload-undo-button-margin-bottom: 0;\n  --va-file-upload-undo-margin-left: 0.5rem;\n}\n.va-file-upload-undo-progress-bar {\n  position: absolute;\n  top: 0;\n  left: 0;\n  --va-progress-bar-transition: width var(--b06eea0e) linear;\n}\n.va-file-upload-undo {\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  flex-wrap: wrap;\n}\n.va-file-upload-undo__button {\n  line-height: var(--va-file-upload-undo-button-line-height);\n  margin-left: var(--va-file-upload-undo-margin-left);\n  margin-top: var(--va-file-upload-undo-button-margin-top);\n  margin-bottom: var(--va-file-upload-undo-button-margin-bottom);\n}\n.va-file-upload-undo--vertical {\n  justify-content: center;\n  align-items: center;\n  align-content: center;\n  padding: var(--va-file-upload-undo-padding);\n}\n.va-file-upload-undo--vertical .va-file-upload-undo__text {\n  margin: 1rem 0 0;\n  text-align: center;\n}\n.va-file-upload-undo--vertical .va-file-upload-undo__button {\n  margin: 1rem 0 0;\n}";
const _hoisted_1 = { class: "va-file-upload-undo__text" };
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_progress_bar = resolveComponent("va-progress-bar");
  const _component_va_button = resolveComponent("va-button");
  return openBlock(), createElementBlock(Fragment, null, [
    createVNode(_component_va_progress_bar, {
      "model-value": _ctx.progress,
      rounded: false,
      class: "va-file-upload-undo-progress-bar"
    }, null, 8, ["model-value"]),
    createElementVNode("div", {
      class: normalizeClass(["va-file-upload-undo", _ctx.computedClasses])
    }, [
      createElementVNode("span", _hoisted_1, toDisplayString(_ctx.deletedFileMessage), 1),
      createVNode(_component_va_button, {
        class: "va-file-upload-undo__button",
        "aria-label": _ctx.undoButtonText,
        size: "small",
        outline: "",
        onClick: _cache[0] || (_cache[0] = ($event) => _ctx.$emit("recover"))
      }, {
        default: withCtx(() => [
          createTextVNode(toDisplayString(_ctx.undoButtonText), 1)
        ]),
        _: 1
      }, 8, ["aria-label"])
    ], 2)
  ], 64);
}
const _VaFileUploadUndo = /* @__PURE__ */ _export_sfc(__default__, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaFileUploadUndo as _
};
