// vuestic-ui/dist/web-components/src/components/va-rating/VaRating.js
import { _ as _sfc_main } from "./VaRating.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createElementVNode, withKeys, Fragment, renderList, createBlock, mergeProps, withCtx, renderSlot, normalizeProps, guardReactiveProps, createCommentVNode, normalizeStyle, toDisplayString } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-rating-display: flex;\n  /* Number item */\n  --va-rating-number-item-font-size: inherit;\n  --va-rating-number-item-margin: 0.1em;\n  --va-rating-number-item-font-weight: 700;\n  --va-rating-number-item-cursor: pointer;\n  /* Item wrapper */\n  --va-rating-item-wrapper-cursor: pointer;\n}\n.va-rating {\n  display: var(--va-rating-display);\n  font-family: var(--va-font-family);\n}\n.va-rating__number-item {\n  font-size: 100%;\n  overflow: visible;\n  text-transform: none;\n  padding: 0;\n  background-color: transparent;\n  border-style: none;\n  font-size: var(--va-rating-number-item-font-size);\n  margin: var(--va-rating-number-item-margin);\n  font-weight: var(--va-rating-number-item-font-weight);\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  cursor: pointer;\n}\n.va-rating--disabled .va-rating__number-item {\n  cursor: default;\n  opacity: 0.4;\n  pointer-events: none;\n  -webkit-user-select: none;\n     -moz-user-select: none;\n          user-select: none;\n}\n.va-rating--readonly .va-rating__number-item {\n  cursor: default;\n}\n.va-rating__item-wrapper {\n  display: flex;\n  cursor: var(--va-rating-item-wrapper-cursor);\n}\n.va-rating--readonly .va-rating__item-wrapper, .va-rating--disabled .va-rating__item-wrapper {\n  cursor: default;\n}\n.va-rating-item {\n  display: flex;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n}\n.va-rating--disabled .va-rating-item {\n  cursor: default;\n  opacity: 0.4;\n  pointer-events: none;\n  -webkit-user-select: none;\n     -moz-user-select: none;\n          user-select: none;\n}\n.va-rating--disabled .va-rating-item__wrapper {\n  cursor: initial !important;\n}\n.va-rating--readonly .va-rating-item .va-rating-item__wrapper {\n  cursor: initial !important;\n}\n.va-rating__text-wrapper {\n  padding-left: 10px;\n}";
const _hoisted_1 = ["aria-label"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_VaRatingItemNumberButton = resolveComponent("VaRatingItemNumberButton");
  const _component_va_rating_item = resolveComponent("va-rating-item");
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["va-rating", _ctx.rootClass]),
    "aria-label": _ctx.tp(_ctx.$props.ariaLabel, { max: _ctx.$props.max, value: _ctx.$props.modelValue })
  }, [
    createElementVNode("div", {
      class: "va-rating__item-wrapper",
      onKeyup: [
        _cache[0] || (_cache[0] = withKeys(($event) => _ctx.onArrowKeyPress(-1), ["left"])),
        _cache[1] || (_cache[1] = withKeys(($event) => _ctx.onArrowKeyPress(1), ["right"]))
      ],
      onMouseenter: _cache[2] || (_cache[2] = (...args) => _ctx.onMouseEnter && _ctx.onMouseEnter(...args)),
      onMouseleave: _cache[3] || (_cache[3] = (...args) => _ctx.onMouseLeave && _ctx.onMouseLeave(...args))
    }, [
      (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.$props.max, (itemNumber) => {
        return openBlock(), createBlock(_component_va_rating_item, mergeProps({
          key: itemNumber,
          class: "va-rating__item"
        }, _ctx.VaRatingItemProps, {
          "aria-label": _ctx.tp(_ctx.$props.ariaItemLabel, { max: _ctx.$props.max, value: itemNumber }),
          "model-value": _ctx.getItemValue(itemNumber - 1),
          tabindex: _ctx.tabIndexComputed,
          disabled: _ctx.$props.disabled,
          readonly: _ctx.$props.readonly,
          onHover: ($event) => _ctx.isInteractionsEnabled && _ctx.onItemHoveredValueUpdate(itemNumber - 1, $event),
          "onUpdate:modelValue": ($event) => _ctx.isInteractionsEnabled && _ctx.onItemValueUpdate(itemNumber - 1, $event)
        }), {
          default: withCtx(({ value, onClick }) => [
            renderSlot(_ctx.$slots, "item", normalizeProps(guardReactiveProps({ value, onClick, index: itemNumber })), () => [
              _ctx.$props.numbers ? (openBlock(), createBlock(_component_VaRatingItemNumberButton, mergeProps({ key: 0 }, _ctx.VaRatingItemNumberButtonProps, {
                "model-value": value,
                "item-number": itemNumber
              }), null, 16, ["model-value", "item-number"])) : createCommentVNode("", true)
            ])
          ]),
          _: 2
        }, 1040, ["aria-label", "model-value", "tabindex", "disabled", "readonly", "onHover", "onUpdate:modelValue"]);
      }), 128))
    ], 32),
    _ctx.$props.texts && _ctx.$props.texts.length === _ctx.$props.max ? (openBlock(), createElementBlock("span", {
      key: 0,
      class: "va-rating__text-wrapper",
      style: normalizeStyle({ color: _ctx.computedColor })
    }, toDisplayString(_ctx.$props.texts[Math.round(_ctx.visibleValue) - 1]), 5)) : createCommentVNode("", true)
  ], 10, _hoisted_1);
}
const _VaRating = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaRating as _
};
