// vuestic-ui/dist/web-components/src/components/va-parallax/VaParallax.js
import { _ as _sfc_main } from "./VaParallax.vue_vue_type_script_lang.js";
import { openBlock, createElementBlock, normalizeStyle, createElementVNode, renderSlot } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-parallax-display: block;\n  --va-parallax-width: auto;\n  --va-parallax-z-index: 0;\n  /* Image container */\n  --va-parallax-image-container-contain: strict;\n  --va-parallax-image-container-user-select: none;\n  /* Image */\n  --va-parallax-image-will-change: transform;\n  --va-parallax-image-transition: 0.3s opacity linear;\n}\n.va-parallax {\n  display: var(--va-parallax-display);\n  position: relative;\n  overflow: hidden;\n  width: var(--va-parallax-width);\n  z-index: var(--va-parallax-z-index);\n  font-family: var(--va-font-family);\n}\n.va-parallax__image-container {\n  position: absolute;\n  top: 0;\n  left: 0;\n  right: 0;\n  bottom: 0;\n  z-index: 1;\n  contain: var(--va-parallax-image-container-contain);\n  -webkit-user-select: var(--va-parallax-image-container-user-select);\n     -moz-user-select: var(--va-parallax-image-container-user-select);\n          user-select: var(--va-parallax-image-container-user-select);\n}\n.va-parallax__image {\n  position: absolute;\n  bottom: 0;\n  left: 50%;\n  min-width: 100%;\n  max-width: unset;\n  min-height: 100%;\n  display: none;\n  transform: translate(-50%, 0);\n  will-change: var(--va-parallax-image-will-change);\n  transition: var(--va-parallax-image-transition);\n  z-index: 1;\n}\n.va-parallax__item-container {\n  position: absolute;\n  top: 0;\n  left: 0;\n  right: 0;\n  bottom: 0;\n  z-index: 2;\n  contain: strict;\n}";
const _hoisted_1 = { class: "va-parallax__image-container" };
const _hoisted_2 = ["src", "alt"];
const _hoisted_3 = { class: "va-parallax__item-container" };
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", {
    ref: "rootElement",
    class: "va-parallax",
    style: normalizeStyle(_ctx.computedWrapperStyles)
  }, [
    createElementVNode("div", _hoisted_1, [
      createElementVNode("img", {
        ref: "img",
        class: "va-parallax__image",
        src: _ctx.$props.src,
        alt: _ctx.$props.alt,
        style: normalizeStyle(_ctx.computedImgStyles)
      }, null, 12, _hoisted_2)
    ]),
    createElementVNode("div", _hoisted_3, [
      renderSlot(_ctx.$slots, "default")
    ])
  ], 4);
}
const _VaParallax = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaParallax as _
};
