// vuestic-ui/dist/web-components/src/components/va-select/components/VaSelectOptionList/VaSelectOptionList.js
import { _ as _sfc_main } from "./VaSelectOptionList.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, withKeys, withModifiers, Fragment, renderList, toDisplayString, createCommentVNode, createBlock, withCtx, renderSlot, normalizeProps, guardReactiveProps, createVNode, mergeProps } from "vue";
import { _ as _export_sfc } from "../../../../../plugin-vue_export-helper.js";
const _style_0 = ':root,\n:host {\n  --va-select-option-list-width: 100%;\n  --va-select-option-list-max-height: 12.5rem;\n  /* Group name */\n  --va-select-option-list-group-name-padding: 1rem 0.75rem 0.375rem;\n  --va-select-option-list-group-name-font-size: 0.6rem;\n  --va-select-option-list-group-name-color: var(--va-primary);\n  --va-select-option-list-group-name-font-weight: 700;\n  --va-select-option-list-group-name-text-transform: "uppercase";\n  /* Empty block */\n  --va-select-option-list-empty-padding: 0.5rem;\n}\n.va-select-option-list {\n  overflow: auto;\n  width: var(--va-select-option-list-width);\n  max-height: var(--va-select-option-list-max-height);\n  display: flex;\n  flex-direction: column;\n  font-family: var(--va-font-family);\n  scrollbar-color: var(--va-primary) transparent;\n  scrollbar-width: thin;\n}\n.va-select-option-list--empty {\n  padding: var(--va-select-option-list-empty-block-padding);\n}\n.va-select-option-list::-webkit-scrollbar {\n  width: 4px;\n  height: 4px;\n}\n.va-select-option-list::-webkit-scrollbar-track {\n  box-shadow: none;\n  border-radius: 10px;\n}\n.va-select-option-list::-webkit-scrollbar-thumb {\n  background: var(--va-primary);\n  opacity: 0.3;\n  border-radius: 2px;\n}\n.va-select-option-list__group-name {\n  padding: var(--va-select-option-list-group-name-padding);\n  font-size: var(--va-select-option-list-group-name-font-size);\n  color: var(--va-select-option-list-group-name-color);\n  font-weight: var(--va-select-option-list-group-name-font-weight);\n  text-transform: var(--va-select-option-list-group-name-text-transform);\n}';
const _hoisted_1 = ["tabindex"];
const _hoisted_2 = {
  key: 0,
  class: "va-select-option-list__group-name"
};
const _hoisted_3 = {
  key: 0,
  class: "va-select-option-list--empty"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_select_option = resolveComponent("va-select-option");
  const _component_va_virtual_scroller = resolveComponent("va-virtual-scroller");
  return openBlock(), createElementBlock("div", {
    ref: "root",
    class: "va-select-option-list",
    tabindex: _ctx.tabindex,
    onKeydown: [
      _cache[0] || (_cache[0] = withKeys(withModifiers((...args) => _ctx.focusPreviousOption && _ctx.focusPreviousOption(...args), ["stop", "prevent"]), ["up"])),
      _cache[1] || (_cache[1] = withKeys(withModifiers((...args) => _ctx.focusPreviousOption && _ctx.focusPreviousOption(...args), ["stop", "prevent"]), ["left"])),
      _cache[2] || (_cache[2] = withKeys(withModifiers((...args) => _ctx.focusNextOption && _ctx.focusNextOption(...args), ["stop", "prevent"]), ["down"])),
      _cache[3] || (_cache[3] = withKeys(withModifiers((...args) => _ctx.focusNextOption && _ctx.focusNextOption(...args), ["stop", "prevent"]), ["right"])),
      _cache[4] || (_cache[4] = withKeys(withModifiers((...args) => _ctx.selectHoveredOption && _ctx.selectHoveredOption(...args), ["stop", "prevent"]), ["enter"])),
      _cache[5] || (_cache[5] = withKeys(withModifiers((...args) => _ctx.selectHoveredOption && _ctx.selectHoveredOption(...args), ["stop", "prevent"]), ["space"]))
    ],
    onScrollPassive: _cache[6] || (_cache[6] = (...args) => _ctx.onScroll && _ctx.onScroll(...args))
  }, [
    (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.optionGroups, (options, groupName) => {
      return openBlock(), createElementBlock(Fragment, { key: groupName }, [
        groupName !== "_noGroup" ? (openBlock(), createElementBlock("span", _hoisted_2, toDisplayString(groupName), 1)) : createCommentVNode("", true),
        _ctx.$props.virtualScroller ? (openBlock(), createBlock(_component_va_virtual_scroller, {
          key: 1,
          ref_for: true,
          ref: "virtualScrollerRef",
          items: options,
          "track-by": _ctx.getTrackBy,
          "wrapper-size": _ctx.rootHeight,
          "onScroll:bottom": _ctx.handleScrollToBottom
        }, {
          default: withCtx(({ item: option, index }) => [
            renderSlot(_ctx.$slots, "default", normalizeProps(guardReactiveProps({ option, index, selectOption: _ctx.selectOption })), () => [
              createVNode(_component_va_select_option, mergeProps({
                option,
                "current-option": _ctx.currentOptionComputed,
                disabled: _ctx.getDisabled(option)
              }, _ctx.selectOptionProps, {
                onClick: withModifiers(_ctx.selectHoveredOption, ["stop"]),
                onMouseenter: ($event) => _ctx.handleMouseEnter(option),
                onMousemove: ($event) => _ctx.handleMouseMove(option)
              }), null, 16, ["option", "current-option", "disabled", "onClick", "onMouseenter", "onMousemove"])
            ])
          ]),
          _: 2
        }, 1032, ["items", "track-by", "wrapper-size", "onScroll:bottom"])) : (openBlock(true), createElementBlock(Fragment, { key: 2 }, renderList(options, (option, index) => {
          return renderSlot(_ctx.$slots, "default", normalizeProps(mergeProps({
            key: _ctx.getTrackBy(option)
          }, { option, index, selectOption: _ctx.selectOption })), () => [
            createVNode(_component_va_select_option, mergeProps({
              ref_for: true,
              ref: _ctx.setItemRef(_ctx.getTrackBy(option)),
              "current-option": _ctx.currentOptionComputed,
              option,
              disabled: _ctx.getDisabled(option)
            }, _ctx.selectOptionProps, {
              onClick: withModifiers(_ctx.selectHoveredOption, ["stop"]),
              onMouseenter: ($event) => _ctx.handleMouseEnter(option),
              onMousemove: ($event) => _ctx.handleMouseMove(option)
            }), null, 16, ["current-option", "option", "disabled", "onClick", "onMouseenter", "onMousemove"])
          ]);
        }), 128))
      ], 64);
    }), 128)),
    !_ctx.filteredOptions.length ? (openBlock(), createElementBlock("div", _hoisted_3, toDisplayString(_ctx.noOptionsText), 1)) : createCommentVNode("", true)
  ], 40, _hoisted_1);
}
const _VaSelectOptionList = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaSelectOptionList as _
};
