// vuestic-ui/dist/web-components/src/components/va-infinite-scroll/VaInfiniteScroll.js
import { _ as _sfc_main } from "./VaInfiniteScroll.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createBlock, resolveDynamicComponent, normalizeClass, withCtx, renderSlot, createElementVNode, createVNode, createCommentVNode } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-infinite-scroll-display: flex;\n  --va-infinite-scroll-flex-direction: column;\n  --va-infinite-scroll-spinner-default-width: 100%;\n  --va-infinite-scroll-spinner-default-min-height: 70px;\n  /* Reversed */\n  --va-infinite-scroll-reversed-flex-direction: column-reverse;\n}\n.va-infinite-scroll {\n  display: var(--va-infinite-scroll-display);\n  flex-direction: var(--va-infinite-scroll-flex-direction);\n  font-family: var(--va-font-family);\n}\n.va-infinite-scroll--reversed {\n  flex-direction: var(--va-infinite-scroll-reversed-flex-direction);\n}\n.va-infinite-scroll__spinner {\n  display: flex;\n  justify-content: center;\n  align-items: center;\n}\n.va-infinite-scroll__spinner__default {\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  width: var(--va-infinite-scroll-spinner-default-width);\n  min-height: var(--va-infinite-scroll-spinner-default-min-height);\n}\n.va-infinite-scroll__spinner--invisible {\n  visibility: hidden !important;\n}";
const _hoisted_1 = { class: "va-infinite-scroll__spinner__default" };
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_progress_circle = resolveComponent("va-progress-circle");
  return openBlock(), createBlock(resolveDynamicComponent(_ctx.$props.tag), {
    ref: "element",
    role: "feed",
    class: normalizeClass(["va-infinite-scroll", { "va-infinite-scroll--reversed": _ctx.$props.reverse }]),
    "aria-busy": _ctx.fetching
  }, {
    default: withCtx(() => [
      renderSlot(_ctx.$slots, "default"),
      createElementVNode("div", {
        ref: "spinnerSlotContainer",
        class: normalizeClass(["va-infinite-scroll__spinner", { "va-infinite-scroll__spinner--invisible": !_ctx.fetching }])
      }, [
        !_ctx.$props.disabled ? renderSlot(_ctx.$slots, "loading", { key: 0 }, () => [
          createElementVNode("div", _hoisted_1, [
            createVNode(_component_va_progress_circle, {
              size: "small",
              thickness: 0.15,
              color: _ctx.spinnerColor,
              indeterminate: ""
            }, null, 8, ["thickness", "color"])
          ])
        ]) : createCommentVNode("", true)
      ], 2)
    ]),
    _: 3
  }, 8, ["class", "aria-busy"]);
}
const _VaInfiniteScroll = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaInfiniteScroll as _
};
