// vuestic-ui/dist/web-components/src/components/va-file-upload/VaFileUploadSingleItem/VaFileUploadSingleItem.js
import { _ as _sfc_main } from "./VaFileUploadSingleItem.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createBlock, withCtx, createVNode, createElementVNode, toDisplayString, createTextVNode, createCommentVNode } from "vue";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.js";
const _style_0 = ".va-file-upload-single-item {\n  width: 100%;\n}\n.va-file-upload-single-item__content {\n  display: flex;\n  flex-direction: row;\n  align-items: center;\n  max-width: 100%;\n  width: 100%;\n}\n.va-file-upload-single-item__name {\n  margin-right: 0.25rem;\n  max-width: 80%;\n  overflow: hidden;\n  text-overflow: ellipsis;\n  white-space: nowrap;\n  line-height: normal;\n}\n.va-file-upload-single-item__button {\n  margin-top: 0;\n  margin-bottom: 0;\n  font-weight: 700;\n}";
const _hoisted_1 = { class: "va-file-upload-single-item__name" };
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_button = resolveComponent("va-button");
  const _component_va_list_item_section = resolveComponent("va-list-item-section");
  const _component_va_list_item = resolveComponent("va-list-item");
  return openBlock(), createBlock(_component_va_list_item, {
    disabled: _ctx.disabled,
    "aria-disabled": _ctx.disabled,
    class: "va-file-upload-single-item",
    tabindex: "-1"
  }, {
    default: withCtx(() => [
      createVNode(_component_va_list_item_section, { class: "va-file-upload-single-item__content" }, {
        default: withCtx(() => [
          createElementVNode("div", _hoisted_1, toDisplayString(_ctx.file && _ctx.file.name), 1),
          !_ctx.disabled ? (openBlock(), createBlock(_component_va_button, {
            key: 0,
            class: "va-file-upload-single-item__button",
            "aria-label": _ctx.tp(_ctx.$props.ariaRemoveFileLabel),
            size: "small",
            color: "danger",
            flat: "",
            onClick: _cache[0] || (_cache[0] = ($event) => _ctx.$emit("remove"))
          }, {
            default: withCtx(() => [
              createTextVNode(" Delete ")
            ]),
            _: 1
          }, 8, ["aria-label"])) : createCommentVNode("", true)
        ]),
        _: 1
      })
    ]),
    _: 1
  }, 8, ["disabled", "aria-disabled"]);
}
const _VaFileUploadSingleItem = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaFileUploadSingleItem as _
};
