// vuestic-ui/dist/web-components/src/components/va-carousel/VaCarousel.js
import { _ as _sfc_main } from "./VaCarousel.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, Fragment, withKeys, withModifiers, renderSlot, createVNode, withCtx, createCommentVNode, renderList, mergeProps, normalizeProps, guardReactiveProps, createTextVNode, toDisplayString, createElementVNode } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ':root,\n:host {\n  --va-carousel-background: var(--va-background-secondary);\n  --va-carousel-border-radius: var(--va-block-border-radius);\n  --va-carousel-box-shadow: var(--va-box-shadow);\n  --va-carousel-slides-transition: all 0.3s ease-in-out;\n  --va-carousel-indicators-gap: 0.25rem;\n  --va-carousel-min-height: 80px;\n  /** Fake padding for absolute position elements */\n  --va-carousel-padding: 16px;\n}\n@keyframes va-carousel-fade-appear {\n0% {\n    opacity: 0;\n}\n100% {\n    opacity: 1;\n}\n}\n.va-carousel {\n  display: flex;\n  width: 100%;\n  height: 100%;\n  max-height: 100%;\n  min-height: var(--va-carousel-min-height);\n  background: var(--va-carousel-background);\n  box-shadow: var(--va-carousel-box-shadow);\n  border-radius: var(--va-carousel-border-radius);\n  position: relative;\n  overflow: hidden;\n}\n.va-carousel__content {\n  flex: 1;\n  width: 100%;\n  white-space: nowrap;\n}\n.va-carousel__slides {\n  width: 100%;\n  height: 100%;\n  transition: var(--va-carousel-slides-transition);\n}\n.va-carousel__slide {\n  display: inline-flex;\n  justify-content: center;\n  align-items: center;\n  width: 100%;\n  height: 100%;\n  overflow: hidden;\n}\n.va-carousel__indicators {\n  width: 100%;\n  position: absolute;\n  bottom: var(--va-carousel-padding);\n  display: flex;\n  justify-content: center;\n  z-index: 2;\n  overflow-x: auto;\n  left: 50%;\n  transform: translateX(-50%);\n  scrollbar-color: var(--va-background-element) transparent;\n  scrollbar-width: thin;\n}\n.va-carousel__indicators::-webkit-scrollbar {\n  width: 4px;\n  height: 4px;\n}\n.va-carousel__indicators::-webkit-scrollbar-track {\n  box-shadow: none;\n  border-radius: 10px;\n}\n.va-carousel__indicators::-webkit-scrollbar-thumb {\n  background: var(--va-background-element);\n  opacity: 0.3;\n  border-radius: 2px;\n}\n.va-carousel__indicators > * {\n  margin: 0 var(--va-carousel-indicators-gap);\n}\n.va-carousel__arrow {\n  z-index: 1;\n  width: -moz-max-content;\n  width: max-content;\n  height: -moz-max-content;\n  height: max-content;\n}\n.va-carousel__arrow--right {\n  right: var(--va-carousel-padding);\n  position: absolute;\n  top: 50%;\n  transform: translateY(-50%);\n}\n.va-carousel__arrow--left {\n  left: var(--va-carousel-padding);\n  position: absolute;\n  top: 50%;\n  transform: translateY(-50%);\n}\n.va-carousel--vertical .va-carousel__slide {\n  display: flex;\n}\n.va-carousel--vertical .va-carousel__arrow {\n  z-index: 1;\n}\n.va-carousel--vertical .va-carousel__arrow--right {\n  bottom: var(--va-carousel-padding);\n  top: auto;\n  left: 50%;\n  transform: translateX(-50%);\n}\n.va-carousel--vertical .va-carousel__arrow--left {\n  top: var(--va-carousel-padding);\n  left: 50%;\n  transform: translateX(-50%);\n}\n.va-carousel--vertical .va-carousel__indicators {\n  left: var(--va-carousel-padding);\n  flex-direction: column;\n  width: auto;\n  height: 100%;\n  top: 50%;\n  transform: translateY(-50%);\n  overflow-x: hidden;\n  overflow-y: auto;\n}\n.va-carousel--vertical .va-carousel__indicators > * {\n  margin: var(--va-carousel-indicators-gap) 0;\n}\n.va-carousel .va-image {\n  height: 100%;\n  width: 100%;\n}\n.va-carousel .va-button:focus-visible {\n  outline: 2px solid var(--va-focus) !important;\n  border-radius: "inherit";\n  outline-offset: -2px;\n}';
const _hoisted_1 = ["aria-label"];
const _hoisted_2 = {
  key: 1,
  class: "va-carousel__indicators"
};
const _hoisted_3 = { class: "va-carousel__content" };
const _hoisted_4 = ["aria-hidden", "aria-current", "aria-label"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_button = resolveComponent("va-button");
  const _component_va_hover = resolveComponent("va-hover");
  const _component_va_image = resolveComponent("va-image");
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["va-carousel", {
      "va-carousel--vertical": _ctx.$props.vertical,
      [`va-carousel--${_ctx.$props.effect}`]: true
    }]),
    style: normalizeStyle({ height: _ctx.ratio ? "auto" : _ctx.height }),
    role: "region",
    "aria-label": _ctx.tp(_ctx.$props.ariaLabel)
  }, [
    _ctx.$props.arrows ? (openBlock(), createElementBlock(Fragment, { key: 0 }, [
      _ctx.doShowPrevButton ? (openBlock(), createElementBlock("div", {
        key: 0,
        class: "va-carousel__arrow va-carousel__arrow--left",
        onClick: _cache[0] || (_cache[0] = (...args) => _ctx.prev && _ctx.prev(...args)),
        onKeydown: _cache[1] || (_cache[1] = withKeys(withModifiers((...args) => _ctx.prev && _ctx.prev(...args), ["stop"]), ["enter"]))
      }, [
        renderSlot(_ctx.$slots, "prev-arrow", {}, () => [
          createVNode(_component_va_hover, { stateful: "" }, {
            default: withCtx(({ hover }) => [
              createVNode(_component_va_button, {
                color: hover ? _ctx.computedHoverColor : _ctx.computedColor,
                icon: _ctx.vertical ? "va-arrow-up" : "va-arrow-left",
                "aria-label": _ctx.tp(_ctx.$props.ariaPreviousLabel)
              }, null, 8, ["color", "icon", "aria-label"])
            ]),
            _: 1
          })
        ])
      ], 32)) : createCommentVNode("", true),
      _ctx.doShowNextButton ? (openBlock(), createElementBlock("div", {
        key: 1,
        class: "va-carousel__arrow va-carousel__arrow--right",
        onClick: _cache[2] || (_cache[2] = (...args) => _ctx.next && _ctx.next(...args)),
        onKeydown: _cache[3] || (_cache[3] = withKeys(withModifiers((...args) => _ctx.next && _ctx.next(...args), ["stop"]), ["enter"]))
      }, [
        renderSlot(_ctx.$slots, "next-arrow", {}, () => [
          createVNode(_component_va_hover, { stateful: "" }, {
            default: withCtx(({ hover }) => [
              createVNode(_component_va_button, {
                color: hover ? _ctx.computedHoverColor : _ctx.computedColor,
                icon: _ctx.vertical ? "va-arrow-down" : "va-arrow-right",
                "aria-label": _ctx.tp(_ctx.$props.ariaNextLabel)
              }, null, 8, ["color", "icon", "aria-label"])
            ]),
            _: 1
          })
        ])
      ], 32)) : createCommentVNode("", true)
    ], 64)) : createCommentVNode("", true),
    _ctx.$props.indicators ? (openBlock(), createElementBlock("div", _hoisted_2, [
      (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.$props.items, (item, index) => {
        return openBlock(), createElementBlock("div", mergeProps({
          class: ["va-carousel__indicator", { "va-carousel__indicator--active": index === _ctx.modelValue }],
          key: index
        }, _ctx.indicatorTrigger === "hover" ? { onmouseover: () => _ctx.goTo(index) } : { onclick: () => _ctx.goTo(index) }), [
          renderSlot(_ctx.$slots, "indicator", normalizeProps(guardReactiveProps({ item, index, goTo: _ctx.goTo, isActive: _ctx.isCurrentSlide(index) })), () => [
            createVNode(_component_va_hover, { stateful: "" }, {
              default: withCtx(({ hover }) => [
                createVNode(_component_va_button, {
                  "aria-label": _ctx.tp(_ctx.$props.ariaGoToSlideLabel, { index: index + 1 }),
                  round: "",
                  color: _ctx.isCurrentSlide(index) ? _ctx.computedActiveColor : hover ? _ctx.computedHoverColor : _ctx.computedColor
                }, {
                  default: withCtx(() => [
                    createTextVNode(toDisplayString(index + 1), 1)
                  ]),
                  _: 2
                }, 1032, ["aria-label", "color"])
              ]),
              _: 2
            }, 1024)
          ])
        ], 16);
      }), 128))
    ])) : createCommentVNode("", true),
    createElementVNode("div", _hoisted_3, [
      createElementVNode("div", {
        ref: "slidesContainer",
        class: "va-carousel__slides",
        style: normalizeStyle(_ctx.computedSlidesStyle),
        role: "list"
      }, [
        (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.slides, (item, index) => {
          return openBlock(), createElementBlock("div", {
            key: item,
            role: "listitem",
            class: "va-carousel__slide",
            style: normalizeStyle(_ctx.slideStyleComputed),
            "aria-hidden": !_ctx.isCurrentSlide(index),
            "aria-current": _ctx.isCurrentSlide(index),
            "aria-label": _ctx.tp(_ctx.$props.ariaSlideOfLabel, { index: index + 1, length: _ctx.slides.length })
          }, [
            renderSlot(_ctx.$slots, "default", normalizeProps(guardReactiveProps({ item, index, goTo: _ctx.goTo, isActive: _ctx.isCurrentSlide(index) })), () => [
              createVNode(_component_va_image, mergeProps(_ctx.vaImageProps, {
                src: _ctx.isObjectSlides ? item.src : item,
                alt: _ctx.isObjectSlides ? item.alt : "",
                draggable: false
              }), null, 16, ["src", "alt"])
            ])
          ], 12, _hoisted_4);
        }), 128))
      ], 4)
    ])
  ], 14, _hoisted_1);
}
const _VaCarousel = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaCarousel as _
};
