// vuestic-ui/dist/web-components/src/components/va-avatar-group/VaAvatarGroup.vue_vue_type_script_lang.js
import { defineComponent, computed } from "vue";
import { V as VaAvatar } from "../va-avatar/index.js";
import { p as pick } from "../../../vendor.js";
import { u as useAvatarProps } from "../va-avatar/hooks/useAvatarProps.js";
import { u as useSizeProps, a as useSize } from "../../composables/useSize.js";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.js";
import { u as useBem } from "../../composables/useBem.js";
const _sfc_main = defineComponent({
  name: "VaAvatarGroup",
  components: {
    VaAvatar
  },
  props: {
    ...useSizeProps,
    ...useComponentPresetProp,
    ...useAvatarProps,
    max: {
      type: Number,
      default: void 0
    },
    vertical: {
      type: Boolean,
      default: false
    },
    options: {
      type: Array,
      default: () => []
    }
  },
  setup(props) {
    const classComputed = useBem("va-avatar-group", () => ({
      ...pick(props, ["vertical"])
    }));
    const maxOptions = computed(() => props.options.slice(0, props.max));
    const visibleItemsCount = computed(() => props.max ? props.max + 1 : 1);
    const restOptionsCount = computed(() => props.options.length - (props.max || 0));
    const { sizeComputed, fontSizeComputed } = useSize(props, "VaAvatarGroup");
    const avatarProps = computed(() => ({
      ...props,
      fontSize: fontSizeComputed.value,
      size: sizeComputed.value
    }));
    return {
      classComputed,
      maxOptions,
      visibleItemsCount,
      restOptionsCount,
      avatarProps
    };
  }
});
export {
  _sfc_main as _
};
