// vuestic-ui/dist/web-components/src/components/va-file-upload/VaFileUploadListItem/VaFileUploadListItem.js
import { _ as _sfc_main } from "./VaFileUploadListItem.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createBlock, normalizeClass, withCtx, createVNode, createElementVNode, createElementBlock, toDisplayString, createCommentVNode, withModifiers, withKeys } from "vue";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-file-upload-list-item-size-text-color: var(--va-secondary);\n  --va-file-upload-list-item-border-radius: 0.375rem;\n  --va-file-upload-list-item-background-color: var(--va-white);\n  --va-file-upload-list-item-box-shadow: var(--va-box-shadow);\n}\n.va-file-upload-list-item {\n  background-color: var(--va-file-upload-list-item-background-color);\n  box-shadow: var(--va-file-upload-list-item-box-shadow, var(--va-block-box-shadow));\n  border-radius: var(--va-file-upload-list-item-border-radius, var(--va-block-border-radius));\n  position: relative;\n  line-height: 1.5rem;\n  padding: 1.125rem 0.5rem 1rem 1rem;\n  max-width: 100%;\n  width: 100%;\n}\n.va-file-upload-list-item + .va-file-upload-list-item {\n  margin-top: 0.5rem;\n}\n.va-file-upload-list-item .va-list-item__inner {\n  padding: 0;\n  overflow: hidden;\n}\n.va-file-upload-list-item__content {\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n}\n.va-file-upload-list-item__name {\n  white-space: nowrap;\n  text-overflow: ellipsis;\n  flex-basis: 60%;\n  overflow: hidden;\n}\n.va-file-upload-list-item__size {\n  color: var(--va-file-upload-list-item-size-text-color);\n}\n.va-file-upload-list-item__delete {\n  font-size: 1.5rem;\n  cursor: pointer;\n  --va-button-size: 1.5rem;\n}\n.va-file-upload-list-item--undo {\n  overflow: hidden;\n  position: relative;\n}\n.va-file-upload-list-item--undo .va-list-item-section {\n  padding: 0;\n}";
const _hoisted_1 = { class: "va-file-upload-list-item__content" };
const _hoisted_2 = {
  key: 0,
  class: "va-file-upload-list-item__name"
};
const _hoisted_3 = { class: "va-file-upload-list-item__size" };
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_file_upload_undo = resolveComponent("va-file-upload-undo");
  const _component_va_list_item_section = resolveComponent("va-list-item-section");
  const _component_va_button = resolveComponent("va-button");
  const _component_va_list_item = resolveComponent("va-list-item");
  return openBlock(), createBlock(_component_va_list_item, {
    class: normalizeClass(["va-file-upload-list-item", _ctx.computedClasses]),
    tabindex: "-1",
    disabled: _ctx.disabled,
    "aria-disabled": _ctx.disabled
  }, {
    default: withCtx(() => [
      _ctx.removed && _ctx.undo ? (openBlock(), createBlock(_component_va_list_item_section, { key: 0 }, {
        default: withCtx(() => [
          createVNode(_component_va_file_upload_undo, { onRecover: _ctx.recoverFile }, null, 8, ["onRecover"])
        ]),
        _: 1
      })) : (openBlock(), createBlock(_component_va_list_item_section, { key: 1 }, {
        default: withCtx(() => [
          createElementVNode("div", _hoisted_1, [
            _ctx.file && _ctx.file.name ? (openBlock(), createElementBlock("div", _hoisted_2, toDisplayString(_ctx.file && _ctx.file.name), 1)) : createCommentVNode("", true),
            createElementVNode("div", _hoisted_3, toDisplayString(_ctx.file && _ctx.file.size), 1),
            !_ctx.disabled ? (openBlock(), createBlock(_component_va_button, {
              key: 1,
              flat: "",
              color: "danger",
              icon: "clear",
              class: "va-file-upload-list-item__delete",
              "aria-label": _ctx.tp(_ctx.$props.ariaRemoveFileLabel),
              onClick: withModifiers(_ctx.removeFile, ["stop"]),
              onKeydown: [
                withKeys(withModifiers(_ctx.removeFile, ["stop"]), ["enter"]),
                withKeys(withModifiers(_ctx.removeFile, ["stop"]), ["space"])
              ],
              onFocus: _ctx.onFocus,
              onBlur: _ctx.onBlur
            }, null, 8, ["aria-label", "onClick", "onKeydown", "onFocus", "onBlur"])) : createCommentVNode("", true)
          ])
        ]),
        _: 1
      }))
    ]),
    _: 1
  }, 8, ["class", "disabled", "aria-disabled"]);
}
const _VaFileUploadListItem = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaFileUploadListItem as _
};
