// vuestic-ui/dist/es/src/components/va-file-upload/VaFileUploadSingleItem/VaFileUploadSingleItem.vue_vue_type_script_lang-a1c176ad.mjs
import { defineComponent } from "vue";
import { V as VaFileUploadKey } from "../types-8c5c5611.mjs";
import { V as VaButton } from "../../va-button/index-b0a9ef70.mjs";
import { V as VaListItem, a as VaListItemSection } from "../../va-list/index-c1a821d7.mjs";
import { u as useTranslation } from "../../../composables/useTranslation-acf7d88d.mjs";
import { u as useStrictInject } from "../../../composables/useStrictInject-08ce5ce6.mjs";
const INJECTION_ERROR_MESSAGE = "The VaFileUploadSingleItem component should be used in the context of VaFileUpload component";
const _sfc_main = defineComponent({
  name: "VaFileUploadSingleItem",
  components: {
    VaButton,
    VaListItem,
    VaListItemSection
  },
  emits: ["remove"],
  props: {
    file: { type: Object, default: null },
    ariaRemoveFileLabel: { type: String, default: "$t:removeFile" }
  },
  setup: () => ({
    ...useTranslation(),
    disabled: useStrictInject(VaFileUploadKey, INJECTION_ERROR_MESSAGE).disabled
  })
});
export {
  _sfc_main as _
};

import '../../../../VaFileUploadSingleItem.css';
