// vuestic-ui/dist/es/src/components/va-date-picker/components/VaDatePickerCell.vue_vue_type_script_lang-edfb02bc.mjs
import { defineComponent, computed } from "vue";
import { u as useColors } from "../../../composables/useColors-f23d24ff.mjs";
import { u as useTextColor } from "../../../composables/useTextColor-f36510ca.mjs";
const _sfc_main = defineComponent({
  name: "VaDatePickerCell",
  props: {
    otherMonth: { type: Boolean, default: false },
    today: { type: Boolean, default: false },
    inRange: { type: Boolean, default: false },
    disabled: { type: Boolean, default: false },
    selected: { type: Boolean, default: false },
    weekend: { type: Boolean, default: false },
    hidden: { type: Boolean, default: false },
    focused: { type: Boolean, default: false },
    highlightWeekend: { type: Boolean, default: false },
    highlightToday: { type: Boolean, default: false },
    readonly: { type: Boolean, default: false },
    color: { type: String, default: "primary" }
  },
  emits: ["click"],
  setup(props, { emit }) {
    const onClick = () => {
      if (!props.disabled) {
        emit("click");
      }
    };
    const { getColor } = useColors();
    const bg = computed(() => getColor(props.color));
    const { textColorComputed } = useTextColor(bg);
    return {
      bg,
      onClick,
      textColorComputed
    };
  }
});
export {
  _sfc_main as _
};

import '../../../../VaDatePickerCell.css';
