// vuestic-ui/dist/web-components/src/components/va-file-upload/VaFileUploadGalleryItem/VaFileUploadGalleryItem.js
import { _ as _sfc_main } from "./VaFileUploadGalleryItem.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createBlock, normalizeClass, withCtx, createVNode, createElementBlock, createCommentVNode, createElementVNode, normalizeStyle, toDisplayString } from "vue";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-file-upload-gallery-item-card-box-shadow: var(--va-box-shadow);\n  --va-file-upload-gallery-item-text: var(--va-text-primary);\n  --va-file-upload-gallery-item-text-hover: var(--va-text-inverted);\n}\n.va-file-upload-gallery-item {\n  display: flex;\n  position: relative;\n  margin-bottom: 1rem;\n  margin-right: 0.5rem;\n  flex-basis: calc(14.2857% - 0.5rem);\n  max-width: calc(14.2857% - 0.5rem);\n  min-width: 8.5714rem;\n  border-radius: 0.375rem;\n  overflow: hidden;\n  width: 100%;\n  align-items: stretch;\n}\n@media (max-width: 991.98px) {\n.va-file-upload-gallery-item {\n    flex-basis: calc(16.667% - 0.5rem);\n    max-width: calc(16.667% - 0.5rem);\n}\n}\n@media (max-width: 767.98px) {\n.va-file-upload-gallery-item {\n    flex-basis: calc(20% - 0.5rem);\n    max-width: calc(20% - 0.5rem);\n}\n}\n@media (max-width: 575.98px) {\n.va-file-upload-gallery-item {\n    flex-basis: calc(50% - 0.5rem);\n    max-width: calc(50% - 0.5rem);\n}\n}\n.va-file-upload-gallery-item:last-of-type {\n  margin-right: 0;\n}\n.va-file-upload-gallery-item:hover .va-file-upload-gallery-item__overlay, .va-file-upload-gallery-item:focus .va-file-upload-gallery-item__overlay, .va-file-upload-gallery-item--focused .va-file-upload-gallery-item__overlay {\n  z-index: 3;\n  opacity: 1;\n}\n.va-file-upload-gallery-item:hover .va-file-upload-gallery-item__name, .va-file-upload-gallery-item:focus .va-file-upload-gallery-item__name, .va-file-upload-gallery-item--focused .va-file-upload-gallery-item__name {\n  color: var(--va-file-upload-gallery-item-text-hover);\n}\n.va-file-upload-gallery-item__overlay {\n  display: flex;\n  position: absolute;\n  height: 100%;\n  width: 100%;\n  top: 0;\n  left: 0;\n  flex-direction: column;\n  padding: 0.5rem;\n  z-index: -1;\n  opacity: 0;\n}\n.va-file-upload-gallery-item__overlay-background {\n  position: absolute;\n  height: 100%;\n  width: 100%;\n  top: 0;\n  left: 0;\n  z-index: -1;\n}\n.va-file-upload-gallery-item__image {\n  width: 100%;\n  box-shadow: var(--va-box-shadow);\n  object-fit: cover;\n  z-index: 1;\n}\n.va-file-upload-gallery-item__name {\n  color: var(--va-file-upload-gallery-item-text);\n  white-space: nowrap;\n  overflow: hidden;\n  text-overflow: ellipsis;\n  font-size: 0.875rem;\n}\n.va-file-upload-gallery-item__delete {\n  cursor: pointer;\n  font-size: 1.5rem;\n  margin-top: auto;\n}\n.va-file-upload-gallery-item--not-image .va-file-upload-gallery-item__overlay {\n  display: flex;\n}\n.va-file-upload-gallery-item--undo .va-list-item__inner {\n  display: flex;\n  align-items: flex-start;\n  position: relative;\n}\n.va-file-upload-gallery-item--undo .va-list-item-section {\n  height: inherit;\n  padding: 0;\n}\n.va-file-upload-gallery-item--undo .va-file-upload-undo {\n  flex: 1;\n}";
const _hoisted_1 = ["src", "alt"];
const _hoisted_2 = { class: "va-file-upload-gallery-item__overlay" };
const _hoisted_3 = ["title"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_file_upload_undo = resolveComponent("va-file-upload-undo");
  const _component_va_list_item_section = resolveComponent("va-list-item-section");
  const _component_va_button = resolveComponent("va-button");
  const _component_va_list_item = resolveComponent("va-list-item");
  return openBlock(), createBlock(_component_va_list_item, {
    class: normalizeClass(["va-file-upload-gallery-item", _ctx.classesComputed]),
    tabindex: "-1",
    disabled: _ctx.disabled,
    "aria-disabled": _ctx.disabled,
    onFocus: _ctx.onFocus,
    onBlur: _ctx.onBlur
  }, {
    default: withCtx(() => [
      _ctx.removed && _ctx.undo ? (openBlock(), createBlock(_component_va_list_item_section, { key: 0 }, {
        default: withCtx(() => [
          createVNode(_component_va_file_upload_undo, {
            vertical: "",
            onRecover: _ctx.recoverImage
          }, null, 8, ["onRecover"])
        ]),
        _: 1
      })) : (openBlock(), createBlock(_component_va_list_item_section, { key: 1 }, {
        default: withCtx(() => [
          _ctx.previewImage ? (openBlock(), createElementBlock("img", {
            key: 0,
            src: _ctx.previewImage,
            alt: _ctx.file.name || "",
            class: "va-file-upload-gallery-item__image"
          }, null, 8, _hoisted_1)) : createCommentVNode("", true),
          createElementVNode("div", _hoisted_2, [
            createElementVNode("div", {
              class: "va-file-upload-gallery-item__overlay-background",
              style: normalizeStyle(_ctx.overlayStylesComputed)
            }, null, 4),
            _ctx.file && _ctx.file.name ? (openBlock(), createElementBlock("div", {
              key: 0,
              class: "va-file-upload-gallery-item__name",
              title: _ctx.file.name,
              style: normalizeStyle({ color: _ctx.textColorComputed })
            }, toDisplayString(_ctx.file.name), 13, _hoisted_3)) : createCommentVNode("", true),
            !_ctx.disabled ? (openBlock(), createBlock(_component_va_button, {
              key: 1,
              flat: "",
              color: "danger",
              icon: "va-delete",
              class: "va-file-upload-gallery-item__delete",
              "aria-label": _ctx.tp(_ctx.$props.ariaRemoveFileLabel),
              onClick: _ctx.removeImage,
              onFocus: _ctx.onFocus,
              onBlur: _ctx.onBlur
            }, null, 8, ["aria-label", "onClick", "onFocus", "onBlur"])) : createCommentVNode("", true)
          ])
        ]),
        _: 1
      }))
    ]),
    _: 1
  }, 8, ["class", "disabled", "aria-disabled", "onFocus", "onBlur"]);
}
const _VaFileUploadGalleryItem = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaFileUploadGalleryItem as _
};
