// vuestic-ui/dist/esm-node/src/components/va-navbar/VaNavbar.vue_vue_type_script_lang.mjs
import { defineComponent, computed } from "vue";
import { u as useFixedBarProps, a as useFixedBar } from "../../composables/useFixedBar.mjs";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.mjs";
import { s as setupScroll } from "../../composables/useScroll.mjs";
import { u as useColors } from "../../composables/useColors.mjs";
import { u as useTextColor } from "../../composables/useTextColor.mjs";
const _sfc_main = defineComponent({
  name: "VaNavbar",
  props: {
    ...useFixedBarProps,
    ...useComponentPresetProp,
    color: { type: String, default: "background-secondary" },
    textColor: { type: String },
    shape: { type: Boolean, default: false }
  },
  setup(props) {
    const { scrollRoot, isScrolledDown } = setupScroll(props.fixed);
    const { fixedBarStyleComputed } = useFixedBar(props, isScrolledDown);
    const { getColor, shiftHSLAColor } = useColors();
    const { textColorComputed } = useTextColor();
    const color = computed(() => getColor(props.color));
    const shapeStyleComputed = computed(() => ({
      borderTopColor: shiftHSLAColor(color.value, { h: -1, s: -11, l: 10 })
    }));
    const computedStyle = computed(() => ({
      ...fixedBarStyleComputed.value,
      backgroundColor: color.value,
      color: textColorComputed.value,
      fill: textColorComputed.value
    }));
    return {
      scrollRoot,
      computedStyle,
      shapeStyleComputed
    };
  }
});
export {
  _sfc_main as _
};
