// vuestic-ui/dist/web-components/src/components/va-list/VaListItem.js
import { _ as _sfc_main } from "./VaListItem.vue_vue_type_script_lang.js";
import { openBlock, createBlock, resolveDynamicComponent, normalizeClass, withCtx, createElementVNode, renderSlot } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-list-width: 100%;\n  /* Label */\n  --va-list-label-text-align: center;\n  --va-list-label-padding: 0.3rem 0;\n  /* Item */\n  --va-list-item-display: flex;\n  --va-list-item-align-items: center;\n  --va-list-item-width: 100%;\n  --va-list-item-height: 100%;\n  /* Item label */\n  --va-list-item-label-color: #34495e;\n  --va-list-item-label-display: -webkit-box;\n  --va-list-item-label-box-orient: vertical;\n  --va-list-item-label-overflow: hidden;\n  --va-list-item-label-line-height: normal;\n  --va-list-item-label-caption-font-size: 85%;\n  --va-list-item-label-caption-color: #babfc2;\n  /* Item Section */\n  --va-list-item-section-display: flex;\n  --va-list-item-section-icon-min-width: 1.5rem;\n  --va-list-item-section-icon-align-items: center;\n  --va-list-item-section-icon-justify-content: center;\n  --va-list-item-section-icon-margin: 0.6rem 0.75rem;\n  --va-list-item-section-icon-font-size: 1.25rem;\n  --va-list-item-section-avatar-min-width: 3rem;\n}\n.va-list-item {\n  display: block;\n  font-family: var(--va-font-family);\n}\n.va-list-item--disabled {\n  cursor: default;\n  opacity: 0.4;\n  pointer-events: none;\n  -webkit-user-select: none;\n     -moz-user-select: none;\n          user-select: none;\n}\n.va-list-item:not(.va-list-item--disabled):focus-visible {\n  outline: 2px solid var(--va-focus) !important;\n  border-radius: 2px;\n  outline-offset: -2px;\n}\n.va-list-item__inner {\n  display: var(--va-list-item-display);\n  align-items: var(--va-list-item-align-items);\n  width: var(--va-list-item-width);\n  height: var(--va-list-item-height);\n}";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createBlock(resolveDynamicComponent(_ctx.tagComputed), {
    class: normalizeClass(["va-list-item", _ctx.computedClass]),
    role: "listitem",
    href: _ctx.hrefComputed,
    target: _ctx.target,
    to: _ctx.to,
    replace: _ctx.replace,
    exact: _ctx.exact,
    "active-class": _ctx.activeClass,
    "exact-active-class": _ctx.exactActiveClass,
    tabindex: _ctx.tabIndexComputed
  }, {
    default: withCtx(() => [
      createElementVNode("div", {
        class: "va-list-item__inner",
        onClick: _cache[0] || (_cache[0] = ($event) => _ctx.$emit("click")),
        onFocus: _cache[1] || (_cache[1] = ($event) => _ctx.$emit("focus"))
      }, [
        renderSlot(_ctx.$slots, "default")
      ], 32)
    ]),
    _: 3
  }, 8, ["href", "target", "to", "replace", "exact", "active-class", "exact-active-class", "class", "tabindex"]);
}
const _VaListItem = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaListItem as _
};
