// vuestic-ui/dist/web-components/src/components/va-breadcrumbs/VaBreadcrumbsItem/VaBreadcrumbsItem.js
import { _ as _sfc_main } from "./VaBreadcrumbsItem.vue_vue_type_script_lang.js";
import { openBlock, createBlock, resolveDynamicComponent, normalizeClass, withCtx, renderSlot, createTextVNode, toDisplayString } from "vue";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  /* General */\n  --va-breadcrumb-item-display: inline-flex;\n  --va-breadcrumb-item-color: inherit;\n  /* Hovered */\n  --va-breadcrumb-item-hover-opacity: 0.7;\n  /* Focused */\n  --va-breadcrumb-item-focus-text-decoration: underline;\n}\n.va-breadcrumb-item {\n  display: var(--va-breadcrumb-item-display);\n  color: var(--va-breadcrumb-item-color);\n}\n.va-breadcrumb-item--link {\n  color: inherit;\n}\n.va-breadcrumb-item--link:hover {\n  opacity: var(--va-breadcrumb-item-hover-opacity);\n}\n.va-breadcrumb-item--link:focus {\n  -webkit-text-decoration: var(--va-breadcrumb-item-focus-text-decoration);\n          text-decoration: var(--va-breadcrumb-item-focus-text-decoration);\n}";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createBlock(resolveDynamicComponent(_ctx.tagComputed), {
    class: normalizeClass(["va-breadcrumb-item", _ctx.classComputed]),
    "active-class": _ctx.$props.activeClass,
    href: _ctx.hrefComputed,
    to: _ctx.$props.to,
    target: _ctx.$props.target,
    replace: _ctx.$props.replace,
    append: _ctx.$props.append,
    exact: _ctx.$props.exact,
    "exact-active-class": _ctx.$props.exactActiveClass
  }, {
    default: withCtx(() => [
      renderSlot(_ctx.$slots, "default", {}, () => [
        createTextVNode(toDisplayString(_ctx.label), 1)
      ])
    ]),
    _: 3
  }, 8, ["class", "active-class", "href", "to", "target", "replace", "append", "exact", "exact-active-class"]);
}
const _VaBreadcrumbsItem = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaBreadcrumbsItem as _
};
