// vuestic-ui/dist/web-components/src/components/va-file-upload/VaFileUpload.js
import { _ as _sfc_main } from "./VaFileUpload.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, renderSlot, createElementVNode, toDisplayString, createCommentVNode, createVNode, withCtx, createTextVNode, createBlock, mergeProps } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ':root,\n:host {\n  --va-file-upload-position: relative;\n  --va-file-upload-margin: 0.5rem 0;\n  --va-file-upload-list-margin-top: 1rem;\n  --va-file-upload-dropzone-border-radius: 0.375rem;\n  --va-file-upload-dropzone-cursor: pointer;\n  --va-file-upload-dropzone-text-padding-sm: 0 0 1rem;\n  --va-file-upload-dropzone-field-padding: 1.5rem 2rem;\n  --va-file-upload-dropzone-field-padding-sm: 1.5rem 1rem;\n  --va-file-upload-dropzone-field-button-margin: 0;\n  --va-file-upload-dropzone-field-button-z-index: 10;\n  --va-file-upload-dropzone-field-text-pr: 10px;\n  --va-file-upload-dropzone-list-padding: 0 2rem 1rem;\n}\n.va-file-upload {\n  position: var(--va-file-upload-position);\n  font-family: var(--va-font-family);\n  margin: var(--va-file-upload-margin);\n}\n.va-file-upload .va-file-upload-list {\n  margin-top: var(--va-file-upload-list-margin-top);\n}\n.va-file-upload__field {\n  overflow: hidden;\n  display: flex;\n  align-items: center;\n  position: relative;\n}\n.va-file-upload__field__button {\n  margin: var(--va-file-upload-dropzone-field-button-margin);\n  z-index: 10;\n}\n.va-file-upload__field__text {\n  padding-right: var(--va-file-upload-dropzone-field-text-pr);\n}\n.va-file-upload__field__input {\n  position: absolute;\n  top: 0;\n  right: 0;\n  height: 100%;\n  width: 100%;\n  color: transparent;\n  opacity: 0;\n  cursor: pointer;\n}\n.va-file-upload__field__input::-webkit-file-upload-button {\n  cursor: pointer;\n}\n.va-file-upload--dropzone {\n  background-color: var(--va-file-upload-dropzone-background-color);\n  overflow: hidden;\n  border-radius: var(--va-file-upload-dropzone-border-radius);\n  cursor: var(--va-file-upload-dropzone-cursor);\n}\n.va-file-upload--dropzone .va-file-upload__field {\n  justify-content: center;\n  display: flex;\n  align-items: center;\n  padding: var(--va-file-upload-dropzone-field-padding);\n  transition: height 0.2s;\n  overflow: visible;\n  flex-wrap: wrap;\n}\n@media (max-width: 767.98px) {\n.va-file-upload--dropzone .va-file-upload__field {\n    flex-direction: column;\n    padding: var(--va-file-upload-dropzone-field-padding-sm);\n}\n.va-file-upload--dropzone .va-file-upload__field__text {\n    padding: var(--va-file-upload-dropzone-text-padding-sm);\n    text-align: center;\n}\n}\n.va-file-upload--dropzone .va-file-upload-list {\n  padding: var(--va-file-upload-dropzone-list-padding);\n  margin-top: 0;\n}\n.va-file-upload--disabled .va-file-upload__field__input {\n  cursor: default;\n}\n.va-file-upload--disabled .va-file-upload__field__input::-webkit-file-upload-button {\n  cursor: inherit;\n}\n.va-file-upload .va-button:focus-visible {\n  outline: 2px solid var(--va-focus) !important;\n  border-radius: "inherit";\n  outline-offset: -2px;\n}';
const _hoisted_1 = { class: "va-file-upload__field" };
const _hoisted_2 = {
  key: 0,
  class: "va-file-upload__field__text"
};
const _hoisted_3 = ["accept", "multiple", "disabled"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_button = resolveComponent("va-button");
  const _component_va_file_upload_list = resolveComponent("va-file-upload-list");
  const _component_va_modal = resolveComponent("va-modal");
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["va-file-upload", _ctx.computedClasses]),
    style: normalizeStyle(_ctx.computedStyle)
  }, [
    renderSlot(_ctx.$slots, "default", {}, () => [
      createElementVNode("div", _hoisted_1, [
        _ctx.dropzone ? (openBlock(), createElementBlock("div", _hoisted_2, toDisplayString(_ctx.tp(_ctx.dropZoneText)), 1)) : createCommentVNode("", true),
        createVNode(_component_va_button, {
          class: "va-file-upload__field__button",
          disabled: _ctx.disabled,
          "aria-disabled": _ctx.disabled,
          color: _ctx.colorComputed,
          style: normalizeStyle({ "pointer-events": _ctx.dropzoneHighlight ? "none" : void 0 }),
          onChange: _ctx.changeFieldValue,
          onClick: _ctx.callFileDialogue
        }, {
          default: withCtx(() => [
            createTextVNode(toDisplayString(_ctx.tp(_ctx.uploadButtonText)), 1)
          ]),
          _: 1
        }, 8, ["disabled", "aria-disabled", "color", "style", "onChange", "onClick"])
      ])
    ]),
    createElementVNode("input", {
      ref: "fileInputRef",
      type: "file",
      class: "va-file-upload__field__input",
      tabindex: -1,
      "aria-hidden": "true",
      accept: _ctx.fileTypes,
      multiple: _ctx.type !== "single",
      disabled: _ctx.disabled,
      onChange: _cache[0] || (_cache[0] = (...args) => _ctx.changeFieldValue && _ctx.changeFieldValue(...args)),
      onDragenter: _cache[1] || (_cache[1] = ($event) => _ctx.dropzoneHighlight = true),
      onDragleave: _cache[2] || (_cache[2] = ($event) => _ctx.dropzoneHighlight = false)
    }, null, 40, _hoisted_3),
    _ctx.files.length && !_ctx.$props.hideFileList ? (openBlock(), createBlock(_component_va_file_upload_list, mergeProps({ key: 0 }, _ctx.fileUploadListProps, {
      type: _ctx.type,
      files: _ctx.files,
      color: _ctx.colorComputed,
      onRemove: _ctx.removeFile,
      onRemoveSingle: _ctx.removeSingleFile
    }), null, 16, ["type", "files", "color", "onRemove", "onRemoveSingle"])) : createCommentVNode("", true),
    createVNode(_component_va_modal, {
      modelValue: _ctx.modal,
      "onUpdate:modelValue": _cache[3] || (_cache[3] = ($event) => _ctx.modal = $event),
      "hide-default-actions": "",
      title: "File validation",
      message: "File type is incorrect!"
    }, null, 8, ["modelValue"])
  ], 6);
}
const _VaFileUpload = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaFileUpload as _
};
