// vuestic-ui/dist/esm-node/src/components/va-button-group/VaButtonGroup.vue_vue_type_script_lang.mjs
import { defineComponent, computed } from "vue";
import { o as omit } from "../../../vendor.mjs";
import { V as VaButton } from "../va-button/index.mjs";
import { V as VaConfig } from "../va-config/VaConfig.mjs";
import { e as extractComponentProps } from "../../utils/component-options/extract-component-options.mjs";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.mjs";
import { u as useColors } from "../../composables/useColors.mjs";
import { u as useTextColor } from "../../composables/useTextColor.mjs";
import { f as filterComponentProps } from "../../utils/component-options/filter-props.mjs";
import { u as useBem } from "../../composables/useBem.mjs";
const VaButtonProps = omit(extractComponentProps(VaButton), ["block", "gradient"]);
const _sfc_main = defineComponent({
  name: "VaButtonGroup",
  components: { VaConfig },
  props: {
    ...VaButtonProps,
    ...useComponentPresetProp,
    grow: { type: Boolean, default: false },
    gradient: { type: Boolean, default: false }
  },
  setup: (props) => {
    const { getColor, getGradientBackground } = useColors();
    const colorComputed = computed(() => getColor(props.color));
    const { textColorComputed } = useTextColor(colorComputed);
    const filteredProps = filterComponentProps(VaButtonProps);
    const buttonConfig = computed(() => ({
      VaButton: {
        ...filteredProps.value,
        ...props.gradient && {
          color: "#00000000",
          textColor: textColorComputed.value
        }
      }
    }));
    const computedClass = useBem("va-button-group", () => ({
      square: !props.round,
      grow: props.grow,
      small: props.size === "small",
      large: props.size === "large"
    }));
    const backgroundColor = computed(
      () => props.gradient ? getGradientBackground(colorComputed.value) : "transparent"
    );
    return { buttonConfig, computedClass, backgroundColor };
  }
});
export {
  _sfc_main as _
};
