// vuestic-ui/dist/web-components/src/components/va-viewer/VaViewer.js
import { _ as _sfc_main } from "./VaViewer.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, Fragment, createElementVNode, mergeProps, renderSlot, normalizeProps, guardReactiveProps, createCommentVNode, createBlock, Teleport, createVNode } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ":root {\n  --va-viewer-content-overlay-background-color: rgba(0, 0, 0, 0.45);\n  --va-viewer-content-overlay-z-index: var(--va-z-index-teleport-overlay);\n  --va-viewer-content-controls-panel-padding: 0.75rem;\n  --va-viewer-content-controls-panel-justify-content: flex-end;\n  --va-viewer-content-controls-panel-background-color: rgba(0, 0, 0, 0.1);\n  --va-viewer-content-controls-panel-z-index: calc(var(--va-z-index-teleport-overlay) + 50);\n}\n.va-viewer {\n  --va-image-position: relative;\n}\n.va-viewer-content {\n  position: fixed;\n  inset: 0;\n  z-index: var(--va-viewer-content-overlay-z-index);\n  display: flex;\n  justify-content: center;\n  background-color: var(--va-viewer-content-overlay-background-color);\n}\n.va-viewer-content__main-area {\n  --va-image-content-position: relative;\n  display: flex;\n}\n.va-viewer-content__main-area > * {\n  width: auto;\n  max-width: unset;\n}\n.va-viewer-content__controls-panel {\n  position: fixed;\n  top: 0;\n  left: 0;\n  right: 0;\n  z-index: var(--va-viewer-content-controls-panel-z-index);\n  display: flex;\n  justify-content: var(--va-viewer-content-controls-panel-justify-content);\n  padding: var(--va-viewer-content-controls-panel-padding);\n  background-color: var(--va-viewer-content-controls-panel-background-color);\n}";
const _hoisted_1 = { class: "va-viewer-content" };
const _hoisted_2 = {
  ref: "content",
  class: "va-viewer-content__main-area"
};
const _hoisted_3 = {
  ref: "controls",
  class: "va-viewer-content__controls-panel"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_icon = resolveComponent("va-icon");
  return openBlock(), createElementBlock(Fragment, null, [
    createElementVNode("div", mergeProps({ class: "va-viewer" }, _ctx.$attrs, {
      onClick: _cache[0] || (_cache[0] = (...args) => _ctx.handleAnchorClick && _ctx.handleAnchorClick(...args))
    }), [
      renderSlot(_ctx.$slots, "anchor", normalizeProps(guardReactiveProps({ openViewer: _ctx.openViewer }))),
      !_ctx.$slots.anchor ? renderSlot(_ctx.$slots, "default", { key: 0 }) : createCommentVNode("", true)
    ], 16),
    _ctx.isOpened ? (openBlock(), createBlock(Teleport, {
      key: 0,
      to: _ctx.teleportTarget
    }, [
      createElementVNode("div", _hoisted_1, [
        createElementVNode("div", _hoisted_2, [
          !_ctx.$slots.image ? renderSlot(_ctx.$slots, "default", { key: 0 }) : createCommentVNode("", true),
          renderSlot(_ctx.$slots, "image")
        ], 512),
        createElementVNode("div", _hoisted_3, [
          renderSlot(_ctx.$slots, "controls"),
          renderSlot(_ctx.$slots, "close", normalizeProps(guardReactiveProps({ close: _ctx.closeViewer })), () => [
            createElementVNode("button", {
              class: "va-viewer-content__close-button",
              onClick: _cache[1] || (_cache[1] = (...args) => _ctx.closeViewer && _ctx.closeViewer(...args))
            }, [
              createVNode(_component_va_icon, {
                name: "close",
                color: "backgroundPrimary"
              })
            ])
          ])
        ], 512)
      ])
    ], 8, ["to"])) : createCommentVNode("", true)
  ], 64);
}
const _VaViewer = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [_style_0]]]);
export {
  _VaViewer as _
};
